library/dnet.c

```diff
@@ -791,9 +791,6 @@ int dnet_server_send_write(struct dnet_server_send_ctl *send,
 	dnet_trans_create_send_all(s, &ctl);
 	dnet_session_destroy(s);
 
-	if (!err)
-		err = send->write_error;
-
```

Comment:
Because logic around `dnet_server_send_write` expects that it returns error only if it failed to make a write and in case of error this around logic will send reply to a client with this error. This code led to duplicated reply because it made `dnet_server_send_write` to return an error when some detached write was failed, so logic around send a reply with error which repeats a reply sent by complete callback of the write.